editor/editor_help.cpp

```diff
@@ -43,8 +44,53 @@
 
 #define CONTRIBUTE_URL vformat("%s/contributing/documentation/updating_the_class_reference.html", VERSION_DOCS_URL)
 
+// TODO: this is sometimes used directly as doc->something, other times as EditorHelp::get_doc_data(); Might this be a problem?
 DocTools *EditorHelp::doc = nullptr;
 
+static bool _attempt_doc_load(const String &p_class) {
+	// Docgen always happens in the outer-most class: it also generates docs for inner classes.
+	String outer_class = p_class.get_slice(".", 0);
+	if (!ScriptServer::is_global_class(outer_class)) {
+		return false;
+	}
+
+	Ref<Script> script = ResourceLoader::load(ScriptServer::get_global_class_path(outer_class), outer_class);
+	// Check whether docgen happens in ResourceLoader::load() / whether docs are already cached
+	// If not, call reload() which docgens for GDScript
+	// TODO: could be improved with a docgen-only method, since reload() also compiles GDScripts
+	if (script.is_valid() && (script->are_docs_generated() || (script->reload(true) == OK && script->are_docs_generated()))) {
+		Vector<DocData::ClassDoc> docs = script->get_documentation();
+		for (int j = 0; j < docs.size(); j++) {
+			const DocData::ClassDoc &doc = docs.get(j);
+			EditorHelp::get_doc_data()->add_doc(doc);
+		}
+		return true;
+	}
+
+	return false;
+}
+// Removes unecessary prefix from p_class_specifier when within the p_edited_class context
```

Comment:
Spellcheck reports a typo in "unnecessary" :) Also missing an empty line between the methods.